Cython/Compiler/Nodes.py

```diff
@@ -3453,9 +3483,16 @@ def generate_function_header(self, code, with_pymethdef, proto_only=0):
         if entry.scope.is_c_class_scope and entry.name == "__ipow__":
             arg_code_list.append("CYTHON_UNUSED PyObject *unused")
         if sig.has_generic_args:
-            arg_code_list.append(
-                "PyObject *%s, PyObject *%s" % (
-                    Naming.args_cname, Naming.kwds_cname))
+            varargs_args = "PyObject *%s, PyObject *%s" % (
+                    Naming.args_cname, Naming.kwds_cname)
+            if sig.use_fastcall:
+                fastcall_args = "PyObject *const *%s, Py_ssize_t %s, PyObject *%s" % (
+                        Naming.args_cname, Naming.nargs_cname, Naming.kwds_cname)
```

Comment:
I thought about it, but I couldn't find a good way to improve it. I don't like to store the strings inside the signature, since that would separate that part of the C argument list from the rest. Dealing with the complete C argument list inside the signature doesn't work either, as it needs more data than just the signature.

If you have concrete suggestions for improving, please let me know.